### DISCUSSION TITLE
How to add a js file to a single page - and not to the whole website?

### DISCUSSION THREAD

**Discussion by eviau** [sphinx-doc/sphinx#12710] [Q&A]
Hi !

I am working on some code that should be stored in a js file, and added to the website.

Using the `add_js_file` in the setup section of an extension, I am able to add it to every page on the website - a good start !

Doing an extensive search allowed me to discover that I could add it to a single page - https://github.com/sphinx-doc/sphinx/pull/8631/ - or so I believe ? This [pydata theme docs ](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/static_assets.html#use-an-event-to-add-it-to-specific-pages)lets me think so ?

Am I misunderstanding something ? If I am correct, how should I be able to add the js file to a given page ?

Thanks in advance, I can detail what I tried so far if needed :)

**Comment by AA-Turner** [comment]
```python
def insert_javascript_on_xxx_page(app, page_name, template_name, context, doctree):
    if page_name == 'xxx_page':
        app.add_js_file('script.js')


def setup(app):
  app.connect('html-page-context', insert_javascript_on_xxx_page)
```

**Reply by eviau** [reply]
thank you!

and if my understanding is correct, this goes into conf.py and there is no way to use it in an extension ?

**Reply by mgeier** [reply]
Yes, you can put this into your `conf.py`, but you can also use it in an extension!

In fact, defining a `setup` function in `conf.py` is quite advanced, in most cases this is not necessary. However, every extension has a `setup` function, and it is common for extensions to define an event handler for `'html-page-context'`.

**Comment by eviau** [comment]
thanks to @mgeier and @AA-Turner : i should be able to figure something out. closing this for now.
